pallets/ddc-clusters/src/migrations.rs

```diff
@@ -402,3 +405,151 @@ pub mod v2 {
+		fn on_runtime_upgrade() -> Weight {
+			let current_version = Pallet::<T>::current_storage_version();
+			let onchain_version = Pallet::<T>::on_chain_storage_version();
+			let mut weight = T::DbWeight::get().reads(1);
+
+			if onchain_version == 2 && current_version == 3 {
+				let mut nodes_count_by_cluster: BTreeMap<ClusterId, ClusterNodesCount> =
+					BTreeMap::new();
+
+				let mut translated_clusters = 0u64;
+				Clusters::<T>::translate::<OldCluster<T::AccountId>, _>(
+					|cluster_id, old_cluster| {
+						translated_clusters.saturating_inc();
+						nodes_count_by_cluster.insert(cluster_id, 0);
+						Some(old_cluster.migrate_to_v3())
+					},
+				);
+				weight.saturating_accrue(
+					T::DbWeight::get().reads_writes(translated_clusters, translated_clusters),
+				);
+
+				current_version.put::<Pallet<T>>();
+				weight.saturating_accrue(T::DbWeight::get().writes(1));
+				log::info!(
+					target: LOG_TARGET,
+					"Upgraded {} clusters, storage to version {:?}",
+					translated_clusters,
+					current_version
+				);
+
+				// Update storage version.
+				StorageVersion::new(3).put::<Pallet<T>>();
+
+				weight
+			} else {
+				log::info!(
+					target: LOG_TARGET,
+					"Migration did not execute. This probably should be removed"
+				);
+
+				weight
+			}
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+			frame_support::ensure!(
+				Pallet::<T>::on_chain_storage_version() == 1,
+				"must upgrade linearly"
+			);
+			let pre_clusters_count = Clusters::<T>::iter_keys().count();
+			let pre_clusters_nodes_count = ClustersNodes::<T>::iter_keys().count();
+			let pre_clusters_nodes_stats_count = ClustersNodesStats::<T>::iter_keys().count();
+
+			assert_eq!(
+				pre_clusters_nodes_stats_count, 0,
+				"clusters statistics should be empty before the migration"
+			);
+
+			Ok((pre_clusters_count as u32, pre_clusters_nodes_count as u32).encode())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(state: Vec<u8>) -> Result<(), DispatchError> {
+			let (pre_clusters_count, pre_clusters_nodes_count): (u32, u32) = Decode::decode(
+				&mut &state[..],
+			)
+			.expect("the state parameter should be something that was generated by pre_upgrade");
+
+			let post_clusters_count = Clusters::<T>::iter().count() as u32;
+			assert_eq!(
+				pre_clusters_count, post_clusters_count,
+				"the clusters count before (pre: {}) and after (post: {}) the migration should be the same",
+				pre_clusters_count, post_clusters_count
+			);
+			let post_clusters_nodes_count = ClustersNodes::<T>::iter().count() as u32;
+			assert_eq!(
+				pre_clusters_nodes_count, post_clusters_nodes_count,
+				"the clusters nodes count before (pre: {}) and after (post: {})  the migration should be the same", 
+				pre_clusters_nodes_count, post_clusters_nodes_count
+			);
+
+			let post_clusters_nodes_stats_count = ClustersNodesStats::<T>::iter().count() as u32;
+			assert_eq!(
```

Comment:
Same here, you shouldn't check for `post_clusters_nodes_count` and `post_clusters_nodes_count` if you don't modify the related storage items